tool/gen_etld_data.rb

```diff
@@ -7,23 +7,26 @@
 require 'domain_name'
 require 'set'
 require 'erb'
+require 'yaml'
 
 def main
-  dat_file = $basedir + 'data' + 'effective_tld_names.dat'
-  dir      = $basedir + 'lib' + 'domain_name'
-  erb_file = dir + 'etld_data.rb.erb'
-  rb_file  = dir + 'etld_data.rb'
+  dat_file      = $basedir + 'data/' + 'effective_tld_names.dat'
+  yaml_file     = $basedir + 'lib/data/' + 'etld.yaml'
+  marshall_file = $basedir + 'lib/cache/' + 'etld'
```

Comment:
`$basedir` is a Pathname, so you need no trailing slashes.  I'd probably say `$basedir.join('lib', 'cache', 'etld')` instead.